std/algorithm/mutation.d

```diff
@@ -2389,6 +2374,49 @@ void swap(T)(ref T lhs, ref T rhs) if (is(typeof(lhs.proxySwap(rhs))))
     lhs.proxySwap(rhs);
 }
 
+/**
+Swaps two elements in-place of a random access range `r`,
+specified by their indices `i1` and `i2`.
+
+Params:
+    r  = a $(XREF_PACK_NAMED range,primitives,isRandomAccess,random access range)
+        with swappable elements
+    i1 = first index
+    i2 = second index
+*/
+void swapAt(R)(R r, size_t i1, size_t i2)
```

Comment:
I guess it still makes a random-access range, I'd be surprised to see range that swaps with arbitrary indices but provides no RA.

The point is rather to support sealed ranges, that return rvalues from opIndex.